content/300-guides/200-deployment/201-serverless/300-deploy-to-vercel.mdx

```diff
@@ -6,281 +6,47 @@ metaDescription: 'Learn how to deploy a Next.js application based on Prisma Clie
-
-To learn more about how to create migrations with Prisma Migrate, see our [Get Started guide](/getting-started/setup-prisma/start-from-scratch/relational-databases-typescript-postgresql).
-
-## 1. Create the project and deploy to Vercel
-
-To begin, you will deploy the [`deployment-example-vercel`](https://github.com/prisma/deployment-example-vercel) example repository to Vercel. As part of the deployment steps, you will clone the GitHub repository and connect your application to your PostgreSQL database.
-
-### Steps
-
-1. Go to [`deployment-example-vercel`](https://github.com/prisma/deployment-example-vercel) and click the **Deploy** button.
-
-   ![Vercel deploy button](./images/300-10-deploy-to-vercel-deploy-button.png)
-
-   <Admonition type="info">
-
-   The [Deploy Button](https://vercel.com/docs/deploy-button) is a Vercel feature that allows you to deploy a project and clone the source Git repository to your own repository.
-
-   </Admonition>
-
-2. On the **Create Git Repository** screen, select **GitHub**.
-
-   ![Select GitHub on the Create Git Repository screen](./images/300-20-deploy-to-vercel-select-github.png)
-
-3. Pick a repository name. This will be the name of the cloned repository in your GitHub account. In this guide, we use the name `prisma-vercel-example`:
-
-   ![Pick a repository name](./images/300-30-deploy-to-vercel-create-git-repo.png)
-
-   <Admonition type="info">
-
-   If this is your first time using Vercel, you will be prompted to install the Vercel app in your GitHub account.
-
-   </Admonition>
-
-4. Click **Create**.
-
-5. On the **Configure Project** screen, set the `DATABASE_URL` environment variable to the connection string for your PostgreSQL database:
-
-   ![Configure the project](./images/300-40-deploy-to-vercel-configure-project.png)
-
-   The database connection string has the following format:
-
-   ```code
-   postgresql://__USER__:__PASSWORD__@__HOST__/__DATABASE__
-   ```
-
-   For example, on Supabase it looks similar to:
-
-   ```code
-   postgresql://janedoe:randompassword@db.projectcode.supabase.co:5432/postgres
-   ```
-
-6. Click **Deploy**.
+## Generate Prisma during build
 
-   Vercel will now deploy your application. The `package.json` file in the root directory of the example project uses a [`vercel-build` script](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#custom-build-step-for-node.js) to run `prisma generate && prisma migrate deploy && next build`. This generates Prisma Client, applies the migrations, and then builds the Next.js project. This ensures that the generated Prisma Client in `node_modules/@prisma/client` is available to the Next.js serverless functions.
+Vercel will automatically cache dependencies on deployment. For most applications, this will not cause any issues but for Prisma it may result in outdated clients. To avoid this issue, add `prisma generate` as a postinstall step to your application:
```

Comment:
```suggestion
Vercel will automatically cache dependencies on deployment. For most applications, this will not cause any issues. However, for Prisma, it may result in an outdated version of Prisma Client on a change in your Prisma schema. To avoid this issue, add `prisma generate` to the `postinstall` script of your application:
```